src/components/pages/collectRecordFormPages/FishBeltForm/FishBeltForm.js

```diff
@@ -218,29 +233,68 @@ const FishBeltForm = ({ isNewRecord }) => {
     return Promise.resolve()
   }
 
+  const PartiallyAppliedFishBeltObservationsTable = useCallback(
+    (props) => {
+      // the conditional here makes tests happy
+      if (fishNameConstants.length && fishNameOptions.length) {
+        return (
+          <FishBeltObservationTable
+            fishNameConstants={fishNameConstants}
+            fishNameOptions={fishNameOptions}
+            {...props}
+          />
+        )
+      }
+
+      return null
+    },
+
+    [fishNameConstants, fishNameOptions],
+  )
+
+  const PartiallyAppliedFishBeltTransectInputs = useCallback(
+    (props) => {
+      return (
+        <FishBeltTransectInputs
+          observationsState={observationsState}
+          observationsDispatch={observationsDispatch}
```

Comment:
So some of the failing tests I got earlier were because I passed just the reducer, instead of its components (state and dispatch). React was rerendering it too much. React hocus pocus.